content/blog/survival-analysis-parsnip-adjacent/index.Rmd

```diff
@@ -0,0 +1,227 @@
+
+Our focus so far has been on model specification, fitting, and prediction so let's dive in and see what's new!
+
+
+### Model specification
+
+Several engines can be used for multiple purposes with the same model type, e.g. the `rpart` engine for `decision_tree()` can be used for classification or regression. This distinction can be made in parsnip by specifying the mode of a model.
+We have now introduced a new mode `"censored regression"` in parsnip for models which can be used for survival analysis. The aforementioned `decision_tree()` with the `rpart` engine can also be used to fit a survival decision tree:
+
+```{r new-mode}
+library(censored)
+
+decision_tree() %>% 
+  set_engine("rpart") %>% 
+  set_mode("censored regression")
+
+```
+
+You may notice a new package name here, `censored`. The number of models supported in tidymodels is growing. The core models live in parsnip, however there are several parsnip-adjacent extension packages with additional models such as [discrim](https://discrim.tidymodels.org/) for discriminant analysis models, [poissonreg](https://poissonreg.tidymodels.org/) for Poisson regression models, and [baguette](https://baguette.tidymodels.org/) for ensemble models via bagging. The [censored](https://tidymodels.github.io/censored/) package is the extension package for censored regression. 
+
+Previously we kept the code for the model specifications and their engines in the relevant extension package. While this looks like a straightforward solution to keep code modular and maintainable, we are now reaching such richness in available models that we ran into the following: If the model specification for a `bag_tree()` lives in baguette but we want to add an engine for the new "censored regression" mode in censored, censored has to depend on both parsnip and baguette. To avoid such dependency structures, we are now moving towards keeping all model specifications in parsnip and the extensions packages contain the engine code. 
+
+
+### Model fitting
+
+For fitting a model, parsnip typically supports a formula and a matrix interface, regardless of the interface of the engine. For censored regression models, we are currently only supporting a formula interface. Most engines already support this type of interface, and a formula allows us to specify the outcome via the standard `survival::Surv()` function, tying together event time and status. 
+
+
+A formula also let's us specify stratification for a proportional hazards model. For the `survival::coxph()` function, which powers the `survival` engine for `proportional_hazards()`, this may look like 
+
+<div align="center">
+  `Surv(time, event) ~ x + z + strata(s)`.
+</div>
+
+
+The glmnet package, which powers the `glmnet` engine, does not have a formula interface. It requires us to specify the strata by stratifying the response via `stratifySurv(y, s)`. The response is typically a `Surv` object, so you could end up with a formula like this:
```

Comment:
The following sounds a little repetitive
 
> The glmnet package, which powers the `glmnet` engine,

Reply:
```suggestion
Sine other package can fit this model, we can try another; let's use the glmnet package. However, `glmnet::glmnet()`, does not have a formula interface and it requires us to specify the strata by stratifying the response via `stratifySurv(y, s)`. The response is typically a `Surv` object, so you could end up with a formula like this:
```

Reply:
"sine" 🙄